balancer_conn_wrappers.go

```diff
@@ -205,36 +117,24 @@ func (ccb *ccBalancerWrapper) updateSubConnState(sc balancer.SubConn, s connecti
 	if sc == nil {
 		return
 	}
-	ccb.updateCh.Put(&scStateUpdate{
-		sc:    sc,
-		state: s,
-		err:   err,
+	ccb.serializer.Schedule(func(_ context.Context) {
+		ccb.balancer.UpdateSubConnState(sc, balancer.SubConnState{ConnectivityState: s, ConnectionError: err})
 	})
 }
 
-// handleSubConnStateChange handles a SubConnState update from the update
-// channel and invokes the appropriate method on the underlying balancer.
-func (ccb *ccBalancerWrapper) handleSubConnStateChange(update *scStateUpdate) {
-	ccb.balancer.UpdateSubConnState(update.sc, balancer.SubConnState{ConnectivityState: update.state, ConnectionError: update.err})
-}
-
 func (ccb *ccBalancerWrapper) exitIdle() {
-	ccb.updateCh.Put(&exitIdleUpdate{})
-}
-
-func (ccb *ccBalancerWrapper) handleExitIdle() {
-	if ccb.cc.GetState() != connectivity.Idle {
-		return
-	}
-	ccb.balancer.ExitIdle()
+	ccb.serializer.Schedule(func(_ context.Context) {
+		if ccb.cc.GetState() != connectivity.Idle {
```

Comment:
Is this potentially racy?  What if it's pushing a new picker that _is_ IDLE?  Should we just always call ExitIdle maybe and let the LB policy choose whether to ignore it?

Reply:
Makes sense. And the docstring for `ExitIdle` does say this: `ExitIdle instructs the LB policy to reconnect to backends / exit the IDLE state, if appropriate and possible.` So, we can leave it to the LB policy to decide what to do if they are not in IDLE when they receive this call.